rs/sns/cli/src/main.rs

```diff
@@ -30,7 +30,12 @@ async fn main() -> Result<()> {
         SubCommand::List(args) => list::exec(args, &agent).await,
         SubCommand::Health(args) => health::exec(args, &agent).await,
         SubCommand::UpgradeSnsControlledCanister(args) => {
-            upgrade_sns_controlled_canister::exec(args, &agent).await
+            match upgrade_sns_controlled_canister::exec(args, &agent).await {
```

Comment:
I think with a little restructuring, you could have main only pass 'std::env::args()' into a function that parses and runs the function.  

Then you could test, in unit tests, everything else (assuming you have a little bit of dependency injection for things with side effects you don't want to run in tests).  

NCR, obviously, just an observation for future PRs